zebra-chain/src/orchard/shielded_data.rs

```diff
@@ -38,6 +38,29 @@ pub struct ShieldedData {
     pub binding_sig: Signature<Binding>,
 }
 
+impl ShieldedData {
+    /// Iterate over the [`Action`]s for this transaction, returning them as
+    /// their generic type.
+    pub fn actions(&self) -> impl Iterator<Item = &Action> {
+        self.actions.actions()
+    }
+
+    /// Collect the [`Nullifier`]s for this transaction, if it contains
+    /// [`Action`]s.
+    pub fn nullifiers(&self) -> impl Iterator<Item = &Nullifier> {
+        self.actions().map(|action| &action.nullifier)
+    }
+}
+
+impl AtLeastOne<AuthorizedAction> {
+    /// Iterate over the [`Action`]s of the `AuthorizedAction` for this transaction,
+    /// returning them as their generic type.
```

Comment:
1. There are no generics in the iterator item type `&Action`
2. `AtLeastOne<AuthorizedAction>` is a list type, not a single item

```suggestion
    /// Iterate over the [`Action`]s for the [`AuthorizedAction`]s in this transaction.
```

Reply:
This doesn't seem to have been fixed, but @jvff suggested something similar in #2185.